### DISCUSSION TITLE
deployed.pt file size limit after fine tuning?

### DISCUSSION THREAD

**Discussion by lees58915** [MDIL-SNU/SevenNet#125] [General]
After I fine tuned the .pt file with 180 OUTCAR data from VASP, the size of the .pt file increased up to 94GB, and the maximum capacity of our GPU device (multi-GPU can't be used) is 80GB. When I ran the input file `in.lmp`, this error occured

`ERROR: error loading the model, check the path of the model (src/pair_e3gnn.cpp:320)`

But I was able to find the model file in the `lammps_sevenn/src/`. Is there anyway I can make this work? 

`-rw-r--r-- 1 sean2019391 sean2019391 94209800278 Nov 13 11:28 /home/sean2019391/Actual_semin/deployed.pt` indices the .pt file size is 94.2GB.

**Comment by YutackPark** [comment]
> Is there anyway I can make this work?

I'm afraid not. Model has to be loaded to GPU before MD simulation. You will encounter OOM error from GPU, and maybe, it seems the real cause of `ERROR: error loading the model`.

I have some questions regarding the large file size. My heaviest SevenNet model is below 100 Mb, so the 94GB seems strange.

1. Did you fine-tune from 7net-0 model? If then, the file size should be much smaller than 94 GB, and fine-tuning should not increase original file size.

2. If you fine-tuned model other then 7net-0, could you share your input.yaml for trainings?

3. What is size of your `checkpoint_xxx.pth` file? Before getting the `deployed.pt` model.

Additional information such as version of SevenNet or steps to reproduce will be helpful.

**Reply by lees58915** [reply]
1. Did you fine-tune from 7net-0 model? If then, the file size should be much smaller than 94 GB, and fine-tuning should not increase original file size.

I fine-tuned from `7net-0` model with only first OUTCAR file. From the second file all the way up to 181th file, I changed to `checkpoint: 'checkpoint_best.pth'` using job script loop. 

2. If you fine-tuned model other then 7net-0, could you share your input.yaml for trainings?
I fine-tuned `7net-0` only for the first OUTCAR file, and used 'checkpoint_best.pth'` for the other 180 OUTCAR files. 

3. What is size of your checkpoint_xxx.pth file? Before getting the deployed.pt model.
The size of `checkpoint: 'checkpoint_best.pth'` is 10.3 MB

`-rw-rw-r-- 1 sean2019391 sean2019391 10277451 Nov  9 00:18 checkpoint_best.pth`

The file size increased to 94GB after I ran `sevenn_graph_build my_train_data.extxyz 5.0`

Thank you

**Reply by YutackPark** [reply]
Thanks for the reply. The file size (94GB) is about your preprocessed `my_train_data.extxyz`, not model size.

Here's some clarification

`checkpoint_XXX.pth`: Used for continuing training the model, or to get LAMMPS potential using `sevenn_get_model checkpoint_best.pth`.

`XXX.pt` file after `sevenn_graph_build`: Used for preprocessing the data. Not a model. Preprocessed data can be used for later training, by setting `load_trainset_path = {path to the .pt file}`. This is developed to skip computationally heavy graph preprocess step.

`deployed_serial.pt` or `deplyed_parallel`: Used to run LAMMPS simulation with trained potential. Can be obtained using `sevenn_get_model {checkpoint path}.

I recommend to seeing "https://github.com/MDIL-SNU/SevenNet?tab=readme-ov-file#sevenn_get_model". I also found that `sevenn_graph_build` part of our README.md is outdated... I'll update it. Sorry for the confusion.

**Reply by lees58915** [reply]
**Thank you for your reply!**

The OUTCAR files I trained on are snapshots of approximately 200 atoms. It took me about three days to fine-tune the model with 180 OUTCAR files. Is it supposed to take this long? Also, I was only able to train the model using one OUTCAR file at a time. Training with more than one file took significantly longer. Since my OUTCAR files are incomparably larger than the samples in `SevenNet/example_inputs`, is there any way to streamline the training process?

**Reply by YutackPark** [reply]
The time took for training heavily depends on your hardware. Note that GPU is much more faster than GPU. SevenNet-0 is a heavy model with many parameters so it is possible that you couldn't see any progress in your log, if you're using CPU. I recommend you to use GPU if you weren't.

200 atoms per structure is not small, but tolerable. Here's the example: (https://github.com/MDIL-SNU/sevennet_tutorial/blob/main/notebooks/SevenNet_finetune_tutorial.ipynb)
In this fine-tuning tutorial (work in progress), we train SevenNet-0 with 42 graphs x 416 atoms. With my RTX-4060, it took only ~17 sec per epoch, so it is plausible.
